src/esp/gfx/Renderer.cpp

```diff
@@ -144,29 +103,9 @@ struct Renderer::Impl {
         {GL::PixelFormat::DepthComponent, GL::PixelType::Float});
 
     /* Unproject the Z */
-    Containers::ArrayView<const Float> data =
-        Containers::arrayCast<const Float>(depthImage.data());
-    for (std::size_t i = 0; i != data.size(); ++i) {
-      const Float z = data[i];
-
-      /* If a fragment has a depth of 1, it's due to a hole in the mesh. The
-         consumers expect 0 for things that are too far, so be nice to them.
-         We can afford using == for comparison as 1.0f has an exact
-         representation and the depth is cleared to exactly this value. */
-      if (z == 1.0f) {
-        ptr[i] = 0.0f;
-        continue;
-      }
-
-      /* The following is
-
-          (az + b) / (cz + d)
-
-         See the comment in draw() above for details. */
-      ptr[i] =
-          Math::fma(depthUnprojection_[0][0], z, depthUnprojection_[1][0]) /
-          Math::fma(depthUnprojection_[0][1], z, depthUnprojection_[1][1]);
-    }
+    unprojectDepth(depthUnprojection_,
```

Comment:
Why not unproject into ptr to avoid the memcpy?

Reply:
Originally it was like this but then I put the memcpy back when extracting this code to a function. Ultimately the operation *should* be done in-place, with no memcpy, so I future-proofed the function to operate in-place as well. The zero-copy update is waiting in #114.